src/TSTransformer/nodes/jsx/transformJsxChildren.ts

```diff
@@ -1,362 +1,23 @@
-				),
-				elseBody: luau.list.make(createJsxAddKeyChild(id, keyId, valueId)),
-			}),
-		),
-	});
-}
-
-function createJsxAddArrayChildren(
-	id: luau.AnyIdentifier,
-	amtSinceUpdate: number,
-	lengthId: luau.AnyIdentifier,
-	expression: luau.Expression,
-) {
-	const keyId = luau.tempId("k");
-	const valueId = luau.tempId("v");
-	return luau.create(luau.SyntaxKind.ForStatement, {
-		ids: luau.list.make(keyId, valueId),
-		expression,
-		statements: luau.list.make(createJsxAddNumericChild(id, offset(lengthId, amtSinceUpdate), keyId, valueId)),
-	});
-}
-
-function createJsxAddMapChildren(id: luau.AnyIdentifier, expression: luau.Expression) {
-	const keyId = luau.tempId("k");
-	const valueId = luau.tempId("v");
-	return luau.create(luau.SyntaxKind.ForStatement, {
-		ids: luau.list.make(keyId, valueId),
-		expression,
-		statements: luau.list.make(createJsxAddKeyChild(id, keyId, valueId)),
-	});
-}
-
-// ideally, this would be done automatically..
-function countCreateJsxAddChildExpressionUses(
-	isPossiblyUndefinedOrFalse: boolean,
-	isPossiblyTrue: boolean,
-	isPossiblyElement: boolean,
-	isPossiblyArray: boolean,
-	isPossiblyMap: boolean,
-) {
-	let expUses = 0;
-	if (isPossiblyElement) {
-		expUses += 1;
-	}
-	if (isPossiblyArray || isPossiblyMap) {
-		expUses += 1;
-		if (isPossiblyElement) {
-			expUses += 3;
-		}
-	}
-	if ((isPossiblyUndefinedOrFalse || isPossiblyTrue) && (isPossiblyElement || isPossiblyArray || isPossiblyMap)) {
-		expUses += 1;
-	}
-	return expUses;
-}
-
-function createJsxAddChild(
-	state: TransformState,
-	id: luau.AnyIdentifier,
-	amtSinceUpdate: number,
-	lengthId: luau.AnyIdentifier,
-	expression: luau.Expression,
-	type: ts.Type,
-): luau.Statement {
-	const isPossiblyUndefinedOrFalse = isPossiblyType(type, isUndefinedType, isBooleanLiteralType(state, false));
-	const isPossiblyTrue = isPossiblyType(type, isBooleanLiteralType(state, true));
-	const isPossiblyElement = isPossiblyType(type, isRoactElementType(state));
-
-	// if map keys are possibly number, we need to add number keys to the end like arrays
-	let areMapKeysPossiblyNumber = false;
-	const isPossiblyMap = isPossiblyType(type, t => {
-		if (isMapType(state)(t)) {
-			if (!areMapKeysPossiblyNumber) {
-				const typeArguments = getTypeArguments(state, t);
-				if (isPossiblyType(typeArguments[0], isNumberType)) {
-					areMapKeysPossiblyNumber = true;
-				}
+export function transformJsxChildren(state: TransformState, children: ReadonlyArray<ts.JsxChild>) {
+	return ensureTransformOrder(
+		state,
+		children
+			// ignore jsx text that only contains whitespace
+			.filter(v => !ts.isJsxText(v) || !v.containsOnlyTriviaWhiteSpaces)
+			// ignore empty jsx expressions, i.e. `{}`
+			.filter(v => !ts.isJsxExpression(v) || v.expression !== undefined),
+		(state, node) => {
+			if (ts.isJsxText(node)) {
+				return luau.string(fixupWhitespaceAndDecodeEntities(node.text) ?? "");
```

Comment:
Unfortunately, special characters aren't properly escaped in this.
If I enter the text `\\`, I should get two `\` in my element. However, the emit is `"\\"`, resulting in just a single `\` in the element.

Reply:
Adding a string escape replace call